### PR TITLE
Eigency recipe

### PR DESCRIPTION
**Pull request by fbordeu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fbordeu** [comment]
@conda-forge/staged-recipes Hello what is the time delay to get feedback on this PR. I'm not in a rush, Just want to know the time delay. thanks.

**Comment by fbordeu** [comment]
"Note If you have questions or have not heard back for a while, you can notify us by including @conda-forge/staged-recipes in your GitHub message."

Any idea of the time delay

**Comment by carterbox** [comment]
We still have open pull request from 2021.

File: recipes/eigency/meta.yaml
Comment (carterbox) [inline recipes/eigency/meta.yaml:10]:
```suggestion
  url: https://github.com/wouterboomsma/eigency/archive/refs/tags/{{ version }}.tar.gz
```
We have a bot that will update the recipe for new releases, but only if you use jinja here.

File: recipes/eigency/meta.yaml
Comment (carterbox) [inline recipes/eigency/meta.yaml:7]: Do not do this. This version is for the package version only.

If you want to indicate a build variant, i.e. that this version is built against eigen3.2 instead of eigen3.3 or something, then you should use [a custom build string](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#build-number-and-string). BUT, you aren't building against more than one eigen version, so this is not needed.

```suggestion
  version: {{ version }}
```

File: recipes/eigency/meta.yaml
Comment (carterbox) [inline recipes/eigency/meta.yaml:31]: My understanding is the eigency provides a Cython API wrapper around Eigen, so doesn't that mean the end user requires both cython and eigen installed to use it?

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/eigency/meta.yaml
Comment (fbordeu) [reply recipes/eigency/meta.yaml:31]: Cython is needed only during the compilation (it writes c++ from python) so ones the package is generated cython is not needed any more.
Eigen is a header only library, so only needed at compilation. 

This means this package is intended for the developing of other packages.

In my case, I'm working on a end user package (python package accelerated by c++) and I use cython and eigency to pass information efficiently. And eigency is the only dependency that is not present on conda-forge at this moment.

File: recipes/eigency/meta.yaml
Comment (carterbox) [inline recipes/eigency/meta.yaml:30]:
```suggestion
    - numpy
    - eigen {{ eigenversion }}
  run:
    - python
    - {{ pin_compatible('numpy') }}
```
https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/eigency/meta.yaml
Comment (xhochy) [inline recipes/eigency/meta.yaml:2]: This is a pre-release version. We would like to have only releases con the main channel. Has there been a release meanwhile?

File: recipes/eigency/meta.yaml
Comment (fbordeu) [reply recipes/eigency/meta.yaml]: Done I changed the lines for the pinning, Thank for all the comments. (it is the first time building a conda-forge recipe)

**Comment by fbordeu** [comment]
@carterbox all is ready, what do I have to do to finish the procedure for this PR. Thanks for the review and all your hard work.

**Comment by fbordeu** [comment]
@carterbox what can I do to speed up the process?
Thanks

**Comment by fbordeu** [comment]
@carterbox what can I do to speed up the process?
Thanks

**Comment by fbordeu** [comment]
@dopplershift do you know a way to know who I have ping treat this PR

**Comment by carterbox** [comment]
Please mention the appropriate review team so that your PR is tagged as ready for review, not individuals.

@conda-forge/help-python-c, please review

**Comment by fbordeu** [comment]
Hello @conda-forge/staged-recipes waiting for review for a long time.  Thanks

File: recipes/eigency/meta.yaml
Comment (dopplershift) [inline recipes/eigency/meta.yaml:17]: Let's use `script_env` instead of jumping through commandline hoops:
```suggestion
build:
  script_env:
    - EIGEN_INC=${PREFIX}/include/eigen3
  script: {{ PYTHON }} -m pip install --no-deps . -vv
  number: 0
```

File: recipes/eigency/meta.yaml
Comment (dopplershift) [inline recipes/eigency/meta.yaml:28]:
```suggestion
    - numpy
```

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/eigency/meta.yaml
Comment (carterbox) [inline recipes/eigency/meta.yaml:16]:
```suggestion
  script_env:
    - EIGEN_INC=%PREFIX%\Library\include\eigen3  # [win]
    - EIGEN_INC=${PREFIX}/include/eigen3  # [unix]
```

**Comment by carterbox** [comment]
This package is shipping CXX source files with the python module in site-packages. Please strip those from the next release because they are not needed.

**Pull request merged**
